src/ConcordiumNetSdk/Types/AccountAddress.cs

```diff
@@ -79,31 +101,35 @@ public static bool IsValid(string? addressAsBase58String)
         }
     }
 
-    public override string ToString()
+    /// <summary>
+    /// Converts the account address to its corresponding protocol buffer message instance.
+    /// </summary>
+    public Concordium.V2.AccountAddress ToProto()
     {
-        return _formatted;
+        return new Concordium.V2.AccountAddress()
+        {
+            Value = Google.Protobuf.ByteString.CopyFrom(this._value)
+        };
     }
 
     public override bool Equals(object? obj)
     {
-        if (ReferenceEquals(null, obj)) return false;
-        if (ReferenceEquals(this, obj)) return true;
-        if (obj.GetType() != GetType()) return false;
-        return AsString == ((AccountAddress) obj).AsString;
-    }
-
-    public override int GetHashCode()
-    {
-        return AsString.GetHashCode();
+        if (ReferenceEquals(null, obj))
+            return false;
+        if (ReferenceEquals(this, obj))
+            return true;
+        if (obj.GetType() != GetType())
+            return false;
+        return ToString() == ((AccountAddress)obj).ToString();
```

Comment:
Why do we have to compare strings? The 32-byte array is the address. We don't have to convert to strings to compare. I would even argue that is incorrect.

Reply:
It was in the existing codebase, so I assumed it was there for a reason :-) I'll get rid of it.